solutions/USACO dec19-milkvisits-silver.mdx

```diff
@@ -0,0 +1,225 @@
+---
+id: usaco-968
+source: USACO Silver 2019 December
+title: Milk Visits
+author: Qi Wang
+---
+
+[Official Analysis (Offline)](http://www.usaco.org/current/data/sol_milkvisits_silver_dec19.html)
+
+**Time Complexity:** $\mathcal{O}(N + M)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+//Official Solution modified by Qi Wang
+
+#include <bits/stdc++.h>
+using namespace std;
+
+typedef vector<int> vi; 
+ 
+#define FOR(i,a,b) for (int i = (a); i < (b); ++i)
+#define F0R(i,a) FOR(i,0,a)
+#define ROF(i,a,b) for (int i = (b)-1; i >= (a); --i)
+#define R0F(i,a) ROF(i,0,a)
+#define trav(a,x) for (auto& a: x)
+ 
+#define pb push_back
+ 
+void setIO(string name) {
+	ios_base::sync_with_stdio(0); cin.tie(0);
+	freopen((name+".in").c_str(),"r",stdin);
+	freopen((name+".out").c_str(),"w",stdout);
+}
+
+const int MX = 100005;
+
+int N,M,comp[MX],num;
+bool col[MX];
+vi adj[MX];
+ 
+void dfs(int x) {
+	if (comp[x]) return;
+	comp[x] = num;
+	trav(t,adj[x]) if (col[t] == col[x]) dfs(t);
+}
+ 
+int main() {
+	setIO("milkvisits"); cin >> N >> M;
+	string s; cin >> s;
+	FOR(i,1,N+1) col[i] = s[i-1]=='G';
+	F0R(i,N-1) {
+		int A,B; cin >> A >> B;
+		adj[A].pb(B), adj[B].pb(A);
+	}
+
+	FOR(i,1,N+1) if (!comp[i]) {
+		num ++;
+        //Making Connected Components
```

Comment:
Keep it consistent to tabs? I can see it renders properly on the website but might as well make backend consistent too.
```suggestion
		//Making Connected Components
```

Reply:
did not notice that! sry

Reply:
ivery interesting tho because it looks fine on my ide???

Reply:
Usually tabs are 4 spaces, so you can't tell. In github tabs are 8 spaces so there's a very obvious difference.

Reply:
oh interesting